### PR TITLE
Spread crud

### PR DESCRIPTION
**Pull request by ValzM**

### CODE REVIEW

File: app/persistence/spread_repository.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import psycopg2
+import os
+from dotenv import load_dotenv, find_dotenv
+
+load_dotenv(find_dotenv())
```
Comment (Eduardo-Cerqueira) [inline app/persistence/spread_repository.py]: find_dotenv() not usefull empty
remove it

File: app/persistence/spread_repository.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+def create_spread(fertilizer_id, plot_number, date, spread_quantity):
+    """
+    Create a new spread in the database.
+
+    :param (str) fertilizer_id: The ID of the fertilizer.
+    :param (int) plot_number: The plot number.
+    :param (date) date: The date of the spread.
+    :param (int) spread_quantity: The quantity of the spread.
+    """
+    existing_spread = get_spread_by_fertilizer(fertilizer_id)
```
Comment (Eduardo-Cerqueira) [inline app/persistence/spread_repository.py:91]: This is done in the route side.

File: app/persistence/spread_repository.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+def update_spread(fertilizer_id, plot_number, new_date, new_spread_quantity):
+    """
+    Update the spread quantity for a spread.
+
+    :param (str) fertilizer_id: The ID of the fertilizer.
+    :param (int) plot_number: The plot number.
+    :param (date) new_date: The new date of the spread.
+    :param (int) new_spread_quantity: The new quantity of the spread.
+    """
+    existing_spread = get_spread_by_fertilizer(fertilizer_id)
```
Comment (Eduardo-Cerqueira) [inline app/persistence/spread_repository.py:114]: This is done in the route side.

File: app/persistence/spread_repository.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+def partial_update_spread(fertilizer_id, plot_number, new_date, new_spread_quantity):
+    """
+    Partial update spread information by fertilizer_id and plot_id.
+
+    :param (str) fertilizer_id: The UUID of the fertilizer.
+    :param (int) plot_number: The UUID of the plot.
+    :param (date) new_date: The new date of the spread.
+    :param (int) new_spread_quantity: The new spread quantity.
+    """
+    existing_spread = get_spread_by_fertilizer(fertilizer_id)
```
Comment (Eduardo-Cerqueira) [inline app/persistence/spread_repository.py:137]: This is done in the route side.

File: app/persistence/spread_repository.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    conn.close()
+
+
+def delete_spread(fertilizer_id):
+    """
+    Delete spread by fertilizer_id.
+
+    :param (str) fertilizer_id: The UUID of the fertilizer.
+    """
+    existing_spread = get_spread_by_fertilizer(fertilizer_id)
```
Comment (Eduardo-Cerqueira) [inline app/persistence/spread_repository.py:158]: This is done in the route side.

**Review by Eduardo-Cerqueira — CHANGES_REQUESTED** [review]

File: app/persistence/spread_repository.py (hunk 1 above)
Comment (ValzM) [reply app/persistence/spread_repository.py]: Done

File: app/persistence/spread_repository.py (hunk 2 above)
Comment (ValzM) [reply app/persistence/spread_repository.py:91]: I need it here so I don't get an internal server error.

**Review by Eduardo-Cerqueira — APPROVED** [review]

**Pull request merged**